### PR TITLE
[ADHOC] fix(evm): update claims on claim

### PR DESCRIPTION
**Pull request by Quazia**
🚨 Please review the [guidelines for contributing](https://github.com/boostxyz/boost-protocol/blob/main/.github/CONTRIBUTING.md) to this repository.

### Description
Added claim increment on the variable incentives
💔 Thank you!

### CODE REVIEW

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Your PR body must reference a Github issue using a valid keyword, or your PR title must include the internal Boost ticket number, ie: `[BOOST-1234] Innovate like nuts`
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e171472cf1f168248403d250db84aab712710cee
</p>

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Your PR body must reference a Github issue using a valid keyword, or your PR title must include the internal Boost ticket number, ie: `[BOOST-1234] Innovate like nuts`
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e171472cf1f168248403d250db84aab712710cee
</p>

**Review by topocount — APPROVED** [review]

**Review by sammccord — APPROVED** [review]

File: packages/evm/contracts/incentives/ERC20VariableIncentive.sol [hunk 1]
```diff
@@ -51,6 +51,7 @@ contract ERC20VariableIncentive is AERC20VariableIncentive, RBAC {
         reward = reward_;
         limit = limit_;
         totalClaimed = 0;
+        claims += 1;
```
Comment (jonathandiep) [inline packages/evm/contracts/incentives/ERC20VariableIncentive.sol:54]: I think this `claims += 1` is in the wrong place?

Should happen on `claim()`, not `initialize()`

Comment (jonathandiep) [inline packages/evm/contracts/incentives/ERC20VariableIncentive.sol:54]: I think this claims += 1 is in the wrong place?

Should happen on claim(), not initialize()

**Review by jonathandiep — CHANGES_REQUESTED** [review]

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against cd08927ce5841c8ea2f8c4881c9c3119155d3f39
</p>

**Review by jonathandiep — APPROVED** [review]

**Pull request merged**